lib/connect-backbone-to-react.js

```diff
@@ -105,6 +105,10 @@ module.exports = function connectBackboneToReact(
         this.setState(mapModelsToProps(this.models));
       }
 
+      componentWillReceiveProps(nextProps) {
```

Comment:
+1 to using `createNewProps`. That should be the proxy through which all state updates should flow.

I'm wondering what we should do about the new `modelMap`?

Within this function should we

```javascript
this.models = nextProps.models || context.models;
this.createNewProps();
```

Even better, if we can find a way to reduce the duplication of `this.models = nextProps.models || context.models;`

Also I'm curious about the scenario of, initially the HOC gets its `modelsMap` from `context` and then later it gets it from `props`. Should `props` always take priority over `context`, no matter the order?